client/gcp/gcp.go

```diff
@@ -0,0 +1,212 @@
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/bytebase/dbcost/client"
+)
+
+// Client is the client struct
+type Client struct {
+	httpClient http.Client
+}
+
+// NewClient return a client
+func NewClient() Client {
+	return Client{
+		httpClient: http.Client{},
+	}
+}
+
+const rdsServiceID = "9662-B51E-5089"
+
+// this apiKey should be blocked before merging
+const apiKey = "AIzaSyAj36rmGvQ_3kBqd6Lp3nSzPDo2_nD-sdM"
+
+// PriceInfoEndpoint is the endpoint for all Cloud SQL services on GCP
+// For more information, please refer to https://cloud.google.com/billing/v1/how-tos/catalog-api
+var PriceInfoEndpoint = fmt.Sprintf("https://cloudbilling.googleapis.com/v1/services/%s/skus?key=%s", rdsServiceID, apiKey)
+
+type unitPrice struct {
+	CurrencyCode string  `json:"currencyCode"`
+	Unit         string  `json:"units"`
+	Nano         float64 `json:"nanos"`
+}
+
+type tieredRate struct {
+	UnitPrice unitPrice `json:"unitPrice"`
+}
+
+type pricingExpression struct {
+	UsageUnit      string        `json:"usageUnit"`
+	TieredRateList []*tieredRate `json:"tieredRates"`
+}
+
+type pricingInfo struct {
+	PricingExpression pricingExpression `json:"pricingExpression"`
+}
+
+type category struct {
+	ServiceDisplayName string `json:"serviceDisplayName"`
+	ResourceFamily     string `json:"resourceFamily"`
+	ResourceGroup      string `json:"resourceGroup"`
+	UsageType          string `json:"usageType"`
+}
+
+type offer struct {
+	ID                string        `json:"skuId"`
+	Description       string        `json:"description"`
+	Category          category      `json:"category"`
+	PricingInfo       []pricingInfo `json:"pricingInfo"`
+	ServiceRegionList []string      `json:"serviceRegions"`
+}
+
+type pricing struct {
+	OfferList     []*offer `json:"skus"`
+	NextPageToken string   `json:"nextPageToken"`
+}
+
+func httpHelper(nextpageToken string) ([]*offer, error) {
+	endpoint := PriceInfoEndpoint
+	if nextpageToken != "" {
+		endpoint = fmt.Sprintf("%v&pageToken=%v", PriceInfoEndpoint, nextpageToken)
+	}
+
+	req, err := http.NewRequest("GET", endpoint, nil)
+	req.Header.Add("User-Agaent", `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Safari/605.1.15`)
+	req.Header.Add("Connection", `keep-alive`)
+	req.Header.Add("Accept", `text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8`)
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("Fail to fetch the info file, [internal]: %v", err)
+	}
+	if res.StatusCode != 200 {
+		return nil, fmt.Errorf("An http error occur , [internal]: %v", err)
+	}
+
+	data, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, fmt.Errorf("Fail when reading response data , [internal]: %v", err)
+	}
+	var offerList []*offer
+	rawData := &pricing{}
+	if err := json.Unmarshal(data, rawData); err != nil {
+		return nil, fmt.Errorf("Fail when unmarshaling response data , [internal]: %v", err)
+	}
+	offerList = rawData.OfferList
```

Comment:
Do we need to declare offerList in line 98? Should this just be

```
offerList := p.OfferList
```

Or maybe not even declaring this extra variable.